wandb/beta/workflows.py

```diff
@@ -0,0 +1,266 @@
+    name: Optional[str],
+) -> Any:
+    """High-level wrapper to add object(s) to an artifact - calls any of the .add* methods
+    under Artifact depending on the type of object that's passed in. This will probably be moved
+    to the Artifact class in the future.
+
+    Args:
+        artifact: `ArtifactInterface` - most likely a LocalArtifact created with `wandb.Artifact(...)`
+
+        path_or_obj: `Union[str, ArtifactEntry, data_types.WBValue]` - either a str or valid object which
+        indicates what to add to an artifact.
+
+        name: `str` - the name of the object which is added to an artifact.
+
+    Returns:
+        Type[Any] - Union[None, ArtifactManifestEntry, etc]
+
+    """
+    if isinstance(path_or_obj, ArtifactEntry):
+        return artifact.add_reference(path_or_obj, name)
+    elif isinstance(path_or_obj, data_types.WBValue):
+        return artifact.add(path_or_obj, name)
+    elif isinstance(path_or_obj, str):
+        if os.path.isdir(path_or_obj):
+            return artifact.add_dir(path_or_obj)
+        elif os.path.isfile(path_or_obj):
+            return artifact.add_file(path_or_obj)
+        else:
+            with artifact.new_file(name) as f:
+                f.write(json.dumps(path_or_obj, sort_keys=True))
+    else:
+        raise ValueError(
+            f"Expected `path_or_obj` to be instance of `ArtifactEntry`, `WBValue`, or `str, found {type(path_or_obj)}"
+        )
+
+
+def _log_artifact_version(
+    name: str,
+    type: str,
+    entries: Dict[str, Union[str, ArtifactEntry, data_types.WBValue]],
+    aliases: Optional[Union[str, List[str]]] = None,
+    description: Optional[str] = None,
+    metadata: Optional[dict] = None,
+    project: Optional[str] = None,
+    scope_project: Optional[bool] = None,
+    job_type: str = "auto",
+) -> ArtifactInterface:
+    """Creates an artifact, populates it, and logs it with a run.
+    If a run is not present, we create one.
+
+    Args:
+        name: `str` - name of the artifact. If not scoped to a project, name will be suffixed by "-{run_id}".
+        type: `str` - type of the artifact, used in the UI to group artifacts of the same type.
+        entries: `Dict` - dictionary containing the named objects we want added to this artifact.
+        description: `str` - text description of artifact.
+        metadata: `Dict` - users can pass in artifact-specific metadata here, will be visible in the UI.
+        project: `str` - project under which to place this artifact.
+        scope_project: `bool` - if True, we will not suffix `name` with "-{run_id}".
+        job_type: `str` - Only applied if run is not present and we create one.
+            Used to identify runs of a certain job type, i.e "evaluation".
+
+    Returns:
+        ArtifactInterface
+
+    """
+    if wandb.run is None:
+        run = wandb.init(
+            project=project, job_type=job_type, settings=wandb.Settings(silent="true")
+        )
+    else:
+        run = wandb.run
```

Comment:
do you want to do it implicitly or do you want to ask the user to start a new run?
for our other integrations we ask the user for this explicitly. I guess if we do it implicitly we should give the user some indication?

Reply:
I think we want it to work out of the box like a one-liner. So implicit for now, but you make a good point. This is under `wandb.beta.workflows` anyways so we can change it if users aren't comfortable with implicit creation.